test/h5test.h

```diff
@@ -208,10 +212,9 @@ extern pthread_key_t test_thread_info_key_g;
     } while (0)
 #define PASSED()                                                                                             \
     do {                                                                                                     \
-        if (!TEST_EXECUTION_THREADED) {                                                                           \
```

Comment:
Note that these changes from `!TEST_EXECUTION_THREADED` back to `IS_MAIN_TEST_THREAD` **will** break output from the multi-threaded API tests. We should decide whether to merge these changes first and then fix the API tests, or re-design this first.